lib/attrsets.nix

```diff
@@ -225,6 +244,23 @@ rec {
     listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [(nameValuePair name v)] else []) (attrNames set));
 
 
+  /* Like `concatMapAttrs`, but `null`s are filtered out.
+
+     Type:
+       filterMapAttrs ::
+         (String -> a -> null | Attrset)
+         -> AttrSet
+         -> AttrSet
+
+     Example:
+       filterMapAttrs
+         (name: value: if value == "a" then { ${name} = value; } else null)
+         { x = "a"; y = "b"; }
+       => { x = "a"; }
+  */
+  filterMapAttrs = f: concatMapAttrs (name: value: let nvp = f name value; in optionalAttrs (nvp != null) nvp);
```

Comment:
I don't know if the conversion from `null` to `{ }` is enough to warrant a separate function. We don't have a use for it now, and all "hand written" functions can just return `{ }` instead.

Reply:
that's pretty true, It made more sense when it was using lists but now its basically the same